core/src/main/java/com/novoda/noplayer/exoplayer/ExoPlayerTwoFacade.java

```diff
@@ -0,0 +1,317 @@
+        // no-op
+    }
+
+    @Override
+    public void onVideoDisabled(DecoderCounters counters) {
+        // no-op
+    }
+
+    public int getBufferedPercentage() {
+        if (hasPlayer()) {
+            return exoPlayer.getBufferedPercentage();
+        } else {
+            return 0;
+        }
+    }
+
+    public void setPlayWhenReady(boolean playWhenReady) {
+        if (hasPlayer()) {
+            exoPlayer.setPlayWhenReady(playWhenReady);
+        }
+    }
+
+    public void seekTo(long positionMillis) {
+        if (hasPlayer()) {
+            exoPlayer.seekTo(positionMillis);
+        }
+    }
+
+    public void release() {
+        if (hasPlayer()) {
+            exoPlayer.release();
+        }
+    }
+
+    public void prepare(Uri uri, ContentType contentType) {
+        exoPlayer.addListener(listener);
+
+        setPlayWhenReady(true);
+
+        MediaSource mediaSource = mediaSourceFactory.create(contentType, uri, eventListener, mediaSourceEventListener);
+        exoPlayer.prepare(mediaSource, true, false);
+    }
+
+    private final ExoPlayer.EventListener listener = new ExoPlayer.EventListener() {
+
+        @Override
+        public void onTimelineChanged(Timeline timeline, Object manifest) {
+            // no-op
+        }
+
+        @Override
+        public void onTracksChanged(TrackGroupArray trackGroups, TrackSelectionArray trackSelections) {
+            // no-op
+        }
+
+        @Override
+        public void onLoadingChanged(boolean isLoading) {
+            // no-op
+        }
+
+        @Override
+        public void onPlayerStateChanged(boolean playWhenReady, int playbackState) {
+            for (Listener listener : listeners) {
+                listener.onPlayerStateChanged(playWhenReady, playbackState);
+            }
+        }
+
+        @Override
+        public void onPlayerError(ExoPlaybackException error) {
+            for (Listener listener : listeners) {
+                listener.onPlayerError(error);
+            }
+        }
+
+        @Override
+        public void onPositionDiscontinuity() {
+            //no-op
+        }
+    };
+
+    private final ExtractorMediaSource.EventListener eventListener = new ExtractorMediaSource.EventListener() {
+        @Override
+        public void onLoadError(IOException error) {
+            internalErrorListener.onLoadError(error);
+        }
+    };
+
+    private final AdaptiveMediaSourceEventListener mediaSourceEventListener = new AdaptiveMediaSourceEventListener() {
+
+        private Bitrate videoBitrate = Bitrate.fromBitsPerSecond(0);
+        private Bitrate audioBitrate = Bitrate.fromBitsPerSecond(0);
+
+        @Override
+        public void onLoadStarted(DataSpec dataSpec, int dataType, int trackType, Format trackFormat, int trackSelectionReason,
+                                  Object trackSelectionData, long mediaStartTimeMs, long mediaEndTimeMs, long elapsedRealtimeMs) {
+            infoListeners.onLoadStarted(
```

Comment:
I'd prefer to keep this one to one for now. After not working on this for a while it can get super confusing with all of the listeners forwarding to each other.